tap_harvest/sync.py

```diff
@@ -0,0 +1,58 @@
+import singer
+from typing import Dict
+from tap_harvest.streams import STREAMS
+from tap_harvest.client import Client
+
+LOGGER = singer.get_logger()
+
+
+def update_currently_syncing(state: Dict, stream_name: str) -> None:
+    """Update currently_syncing in state and write it."""
+    if not stream_name and singer.get_currently_syncing(state):
+        del state["currently_syncing"]
+    else:
+        singer.set_currently_syncing(state, stream_name)
+    singer.write_state(state)
+
+
+def write_schema(stream, client, streams_to_sync, catalog) -> None:
+    """Collect nested child streams to sync and write schema for selected
+    streams."""
+    if stream.is_selected():
+        stream.write_schema()
+
+    for child in stream.children:
+        child_obj = STREAMS[child](client, catalog.get_stream(child))
+        write_schema(child_obj, client, streams_to_sync, catalog)
+        if child in streams_to_sync:
+            stream.child_to_sync.append(child_obj)
```

Comment:
if we intent to keep a separate file for all the schema related functions, this should probably go in there (also in the template)

Reply:
Agree. Moved it to `scehma.py`